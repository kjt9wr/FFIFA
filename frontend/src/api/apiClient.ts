import axios from "axios";

const apiClient = axios.create({
  baseURL: "http://localhost:5000",
  headers: {
    "Content-Type": "application/json",
  },
});

export default apiClient;

// apiClient.js
apiClient.interceptors.response.use(
  (response) => response,
  (error) => {
    console.error("API call failed:", error);
    // Handle specific error cases
    // if (error.response.status === 401) {
    //   // Unauthorized
    // } else if (error.response.status === 404) {
    //   // Not found
    // }
    return Promise.reject(error);
  }
);
