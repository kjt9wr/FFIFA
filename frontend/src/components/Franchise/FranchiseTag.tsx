import React from "react";
import { Col, Row } from "reactstrap";
import { useFranchiseInfo } from "../../custom-hooks/custom-hooks";
import { POSITION } from "../../utilities/enumerations";
import FranchiseTagColumn from "./FranchiseTagColumn";

/*
 * This page displays the prices of kept players and the corresponding franchise tag prices
 */

const FranchiseTag = () => {
  const franchiseInfo = useFranchiseInfo();

  return (
    <div>
      <h2 className="text-center"> Franchise Tag Prices </h2>
      {franchiseInfo && (
        <Row>
          <Col>
            {franchiseInfo.keptQBs && (
              <FranchiseTagColumn
                keptPlayers={franchiseInfo.keptQBs.slice(0, 10)}
                position={POSITION.QB}
                tagPrice={franchiseInfo.qbFranchisePrice}
              />
            )}
          </Col>
          <Col>
            {franchiseInfo.keptRBs && (
              <FranchiseTagColumn
                keptPlayers={franchiseInfo.keptRBs.slice(0, 10)}
                position={POSITION.RB}
                tagPrice={franchiseInfo.rbFranchisePrice}
              />
            )}
          </Col>
          {franchiseInfo.keptWRs && (
            <FranchiseTagColumn
              keptPlayers={franchiseInfo.keptWRs.slice(0, 10)}
              position={POSITION.WR}
              tagPrice={franchiseInfo.wrFranchisePrice}
            />
          )}
          <Col>
            {franchiseInfo.keptTEs && (
              <FranchiseTagColumn
                keptPlayers={franchiseInfo.keptTEs.slice(0, 10)}
                position={POSITION.TE}
                tagPrice={franchiseInfo.teFranchisePrice}
              />
            )}
          </Col>
        </Row>
      )}
    </div>
  );
};

export default FranchiseTag;
