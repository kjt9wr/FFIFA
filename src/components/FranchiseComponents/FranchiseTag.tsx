import React from "react";
import { Col, Row } from "reactstrap";
import { useFranchiseInfo } from "../../custom-hooks/custom-hooks";
import { POSITION } from "../../utilities/enumerations";
import FranchiseTagColumn from "./FranchiseTagColumn";

/*
 * This page displays the prices of kept players and the corresponding franchise tag prices
 */

const FranchiseTag = () => {
  const franchiseInfo = useFranchiseInfo();

  return (
    <div>
      <h2 className="text-center"> Franchise Tag Prices </h2>
      {franchiseInfo && (
        <Row>
          <Col>
            <FranchiseTagColumn
              keptPlayers={franchiseInfo.keptQBs}
              position={POSITION.QB}
              tagPrice={franchiseInfo.qbFranchisePrice}
            />
          </Col>
          <Col>
            <FranchiseTagColumn
              keptPlayers={franchiseInfo.keptRBs}
              position={POSITION.RB}
              tagPrice={franchiseInfo.rbFranchisePrice}
            />
          </Col>
          <FranchiseTagColumn
            keptPlayers={franchiseInfo.keptWRs}
            position={POSITION.WR}
            tagPrice={franchiseInfo.wrFranchisePrice}
          />
          <Col>
            <FranchiseTagColumn
              keptPlayers={franchiseInfo.keptTEs}
              position={POSITION.TE}
              tagPrice={franchiseInfo.teFranchisePrice}
            />
          </Col>
        </Row>
      )}
    </div>
  );
};

export default FranchiseTag;
