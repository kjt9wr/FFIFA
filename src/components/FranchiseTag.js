import React, { Component } from 'react'
import PositionTag from './reusable/PositionTag'
import * as FranchiseService from '../Services/FranchiseService';
import * as Constants from '../Utilities/Constants';

export default class HomePage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            franchiseInfo: {}
        };
    }
    componentDidMount = () => {
        this.getFranchiseInformation();
    }

    render() {
        const { franchiseInfo } = this.state; 
        return (
            <div >
               <h1 style={{ textAlign: "center"}}>Franchise Tag Prices </h1> 
               <div style={{ display: 'flex' }}>

               <PositionTag keptPlayers = {franchiseInfo.keptQBs} position = {Constants.QB} tagPrice = {franchiseInfo.qbFranchisePrice} />
               <PositionTag keptPlayers = {franchiseInfo.keptRBs} position = {Constants.RB} tagPrice = {franchiseInfo.rbFranchisePrice} />
               <PositionTag keptPlayers = {franchiseInfo.keptWRs} position = {Constants.WR} tagPrice = {franchiseInfo.wrFranchisePrice} />
               <PositionTag keptPlayers = {franchiseInfo.keptTEs} position = {Constants.TE} tagPrice = {franchiseInfo.teFranchisePrice} />
               </div>
            </div>
        )
    }

    getFranchiseInformation = async () => {
        const franchiseInfo = await FranchiseService.getFranchiseTags();
        this.setState({
            franchiseInfo
        })
    }
}

