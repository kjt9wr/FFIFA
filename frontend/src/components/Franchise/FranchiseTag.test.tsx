import { render, screen } from "@testing-library/react";
import FranchiseTag from "./FranchiseTag";
import apiClient from "../../api/apiClient";
import * as FranchiseService from "../../services/franchise.service";
import {
  FRANCHISE_TAG_SAMPLE_DATA,
  KEPT_PLAYERS,
} from "../../services/mock-data/services.mock-data";

describe("franchise tag page", () => {
  it("renders successfully", async () => {
    jest.spyOn(apiClient, "get").mockResolvedValue({ data: KEPT_PLAYERS });
    jest
      .spyOn(FranchiseService, "getFranchiseTagDTO")
      .mockResolvedValue(FRANCHISE_TAG_SAMPLE_DATA);
    const view = render(<FranchiseTag />);

    expect(await screen.findByText("RB franchise price: $75")).toBeTruthy();
    expect(view).toMatchSnapshot();
  });
});
