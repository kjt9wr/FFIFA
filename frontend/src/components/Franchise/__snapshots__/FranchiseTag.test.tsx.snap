// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`franchise tag page renders successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="text-center"
        >
           Franchise Tag Prices 
        </h2>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              style="flex: 1;"
            >
              <h3>
                QB
                 franchise price: $
                75
              </h3>
              <div
                class="table-responsive"
              >
                <table
                  class="table table-sm table-borderless table-striped table-hover"
                >
                  <thead>
                    <tr>
                      <th>
                        Kept Player
                      </th>
                      <th>
                        Original Price
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Jayden Daniels
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        67
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Jalen Hurts
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        37
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Kyler Murray
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        37
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Lamar Jackson
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        37
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Patrick Mahomes
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        37
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              style="flex: 1;"
            >
              <h3>
                RB
                 franchise price: $
                75
              </h3>
              <div
                class="table-responsive"
              >
                <table
                  class="table table-sm table-borderless table-striped table-hover"
                >
                  <thead>
                    <tr>
                      <th>
                        Kept Player
                      </th>
                      <th>
                        Original Price
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Bijan Robinson
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        121
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Saquon Barkley
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        100
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Jahmyr Gibbs
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        98
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Josh Jacobs
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        81
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Derrick Henry
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        79
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139); font-weight: bold;"
                      >
                        Najee Harris
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139); font-weight: bold;"
                      >
                        76
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <div
            style="flex: 1;"
          >
            <h3>
              WR
               franchise price: $
              75
            </h3>
            <div
              class="table-responsive"
            >
              <table
                class="table table-sm table-borderless table-striped table-hover"
              >
                <thead>
                  <tr>
                    <th>
                      Kept Player
                    </th>
                    <th>
                      Original Price
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Drake London
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      64
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      AJ Brown
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      61
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      CeeDee Lamb
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      56
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Tyreek Hill
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      56
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Jamarr Chase
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      52
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Malik Nabers
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      51
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Jalen Coker
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      10
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              style="flex: 1;"
            >
              <h3>
                TE
                 franchise price: $
                75
              </h3>
              <div
                class="table-responsive"
              >
                <table
                  class="table table-sm table-borderless table-striped table-hover"
                >
                  <thead>
                    <tr>
                      <th>
                        Kept Player
                      </th>
                      <th>
                        Original Price
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Brock Bowers
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        37
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        Sam Laporta
                      </td>
                      <td
                        style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                      >
                        12
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="text-center"
      >
         Franchise Tag Prices 
      </h2>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            style="flex: 1;"
          >
            <h3>
              QB
               franchise price: $
              75
            </h3>
            <div
              class="table-responsive"
            >
              <table
                class="table table-sm table-borderless table-striped table-hover"
              >
                <thead>
                  <tr>
                    <th>
                      Kept Player
                    </th>
                    <th>
                      Original Price
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Jayden Daniels
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      67
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Jalen Hurts
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      37
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Kyler Murray
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      37
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Lamar Jackson
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      37
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Patrick Mahomes
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      37
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            style="flex: 1;"
          >
            <h3>
              RB
               franchise price: $
              75
            </h3>
            <div
              class="table-responsive"
            >
              <table
                class="table table-sm table-borderless table-striped table-hover"
              >
                <thead>
                  <tr>
                    <th>
                      Kept Player
                    </th>
                    <th>
                      Original Price
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Bijan Robinson
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      121
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Saquon Barkley
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      100
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Jahmyr Gibbs
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      98
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Josh Jacobs
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      81
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Derrick Henry
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      79
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139); font-weight: bold;"
                    >
                      Najee Harris
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139); font-weight: bold;"
                    >
                      76
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          style="flex: 1;"
        >
          <h3>
            WR
             franchise price: $
            75
          </h3>
          <div
            class="table-responsive"
          >
            <table
              class="table table-sm table-borderless table-striped table-hover"
            >
              <thead>
                <tr>
                  <th>
                    Kept Player
                  </th>
                  <th>
                    Original Price
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    Drake London
                  </td>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    64
                  </td>
                </tr>
                <tr>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    AJ Brown
                  </td>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    61
                  </td>
                </tr>
                <tr>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    CeeDee Lamb
                  </td>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    56
                  </td>
                </tr>
                <tr>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    Tyreek Hill
                  </td>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    56
                  </td>
                </tr>
                <tr>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    Jamarr Chase
                  </td>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    52
                  </td>
                </tr>
                <tr>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    Malik Nabers
                  </td>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    51
                  </td>
                </tr>
                <tr>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    Jalen Coker
                  </td>
                  <td
                    style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                  >
                    10
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            style="flex: 1;"
          >
            <h3>
              TE
               franchise price: $
              75
            </h3>
            <div
              class="table-responsive"
            >
              <table
                class="table table-sm table-borderless table-striped table-hover"
              >
                <thead>
                  <tr>
                    <th>
                      Kept Player
                    </th>
                    <th>
                      Original Price
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Brock Bowers
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      37
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      Sam Laporta
                    </td>
                    <td
                      style="color: rgb(248, 248, 242); background-color: rgb(109, 114, 139);"
                    >
                      12
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
